src/cmark.h

```diff
@@ -466,11 +466,24 @@ CMARK_EXPORT void cmark_consolidate_text_nodes(cmark_node *root);
 CMARK_EXPORT
 cmark_parser *cmark_parser_new(int options);
 
+CMARK_EXPORT
+cmark_parser *cmark_parser_new_into(int options, cmark_node *root);
+
 /** Creates a new parser object with the given memory allocator
  */
 CMARK_EXPORT
 cmark_parser *cmark_parser_new_with_mem(int options, cmark_mem *mem);
 
+/** Creates a new parser object with the given node to use as the root
+ * node of the parsed AST.
+ *
+ * This is useful for creating a single document out of multiple parsed
+ * document fragments.
+ */
```

Comment:
Reading this, I wondered:  what happens if the specified root node already has children?  What happens then?  Do the new nodes get added to the end of the children?  Or do they replace existing children?  Maybe the comment should be explicit about this.

Reply:
They appear after, see `add_child` in `blocks.c`.

I do wish it and `cmark_node_append_child` shared code though to make that clearer, however.